python/cudf/cudf/core/dataframe.py

```diff
@@ -5359,6 +5355,12 @@ def _prepare_for_rowwise_op(self):
 
         filtered = self.select_dtypes(include=[np.number, np.bool])
         common_dtype = np.find_common_type(filtered.dtypes, [])
+        if len(filtered.columns) < len(self.columns):
```

Comment:
Unfortunately `DataFrame.columns` is somewhat expensive to compute. Let's use `df._num_columns` instead.

Reply:
Ditto for `self` :)

Reply:
Yup!